test/integration/test_gateway.go

```diff
@@ -578,39 +574,86 @@ func TestScalingDataPlaneThroughGatewayConfiguration(t *testing.T) {
-		require.Eventually(t, testutils.ControlPlaneHasActiveDeployment(t,
-			GetCtx(),
-			controlplaneNN,
-			clients), testutils.ControlPlaneCondDeadline, testutils.ControlPlaneCondTick)
-
-		t.Logf("verifying the deployment managed by the dataplane is ready and has %d available replicas", replicas)
-		dataplanes := testutils.MustListDataPlanesForGateway(t, GetCtx(), gateway, clients)
-		require.Len(t, dataplanes, 1)
-		dataplane := dataplanes[0]
-		require.Equal(t, *dataplane.Spec.DataPlaneOptions.Deployment.DeploymentOptions.Replicas, replicas)
-		dataplaneNN := client.ObjectKeyFromObject(&dataplane)
-		require.Eventually(t, testutils.DataPlaneHasActiveDeployment(t,
-			GetCtx(),
-			dataplaneNN,
-			&appsv1.Deployment{},
-			client.MatchingLabels{
-				consts.GatewayOperatorManagedByLabel: consts.DataPlaneManagedLabelValue,
+	testCases := []struct {
+		name                       string
+		dataplaneDeploymentOptions operatorv1beta1.DeploymentOptions
+		expectedReplicasCount      int
+	}{
+		{
+			name: "replicas=3",
+			dataplaneDeploymentOptions: operatorv1beta1.DeploymentOptions{
+				Replicas: lo.ToPtr[int32](3),
+			},
+			expectedReplicasCount: 3,
+		},
+		{
+			name: "replicas=0",
+			dataplaneDeploymentOptions: operatorv1beta1.DeploymentOptions{
+				Replicas: lo.ToPtr[int32](0),
+			},
+			expectedReplicasCount: 0,
+		},
+		{
+			name: "replicas=5",
+			dataplaneDeploymentOptions: operatorv1beta1.DeploymentOptions{
+				Replicas: lo.ToPtr[int32](5),
+			},
+			expectedReplicasCount: 5,
+		},
+		{
+			name: "replicas=1",
+			dataplaneDeploymentOptions: operatorv1beta1.DeploymentOptions{
+				Replicas: lo.ToPtr[int32](1),
 			},
-			clients), testutils.DataPlaneCondDeadline, testutils.DataPlaneCondTick)
-		require.Eventually(t, testutils.DataPlaneHasNReadyPods(t, GetCtx(), dataplaneNN, clients, int(replicas)), time.Minute, time.Second)
+			expectedReplicasCount: 1,
+		},
+		{
+			name: "horizontal scaling with minReplicas=3",
+			dataplaneDeploymentOptions: operatorv1beta1.DeploymentOptions{
+				Scaling: &operatorv1beta1.Scaling{
+					HorizontalScaling: &operatorv1beta1.HorizontalScaling{
+						MinReplicas: lo.ToPtr[int32](3),
+						MaxReplicas: 5,
+					},
+				},
+			},
+			expectedReplicasCount: 3,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			deploymentOptions := tc.dataplaneDeploymentOptions
+			gatewayConfiguration, err := GetClients().OperatorClient.ApisV1beta1().GatewayConfigurations(namespace.Name).Get(GetCtx(), gatewayConfigurationName, metav1.GetOptions{})
+			require.NoError(t, err)
+			gatewayConfiguration.Spec.DataPlaneOptions.Deployment.DeploymentOptions = deploymentOptions
+			t.Logf("changing the GatewayConfiguration to change dataplane deploymentOptions to %v", deploymentOptions)
+			_, err = GetClients().OperatorClient.ApisV1beta1().GatewayConfigurations(namespace.Name).Update(GetCtx(), gatewayConfiguration, metav1.UpdateOptions{})
```

Comment:
`Update`s tend to fail sporadically in CI because the client doesn't see the most up to date version of a resource. I don't foresee this here but just to be safe do you mind changing this to `Patch` or wrapping in `.Eventually()` ?